### ISSUE TITLE
Currently, list keys are all mandatory - allows default values for keys. 

### ISSUE THREAD

**Issue by aceelindem** [netmod-wg/yang-next#62]

**Comment by mbj4668** [comment]
See also http://svn.tools.ietf.org/svn/wg/netmod/yang-1.1/issues.html#sec-10

2024 update: "svn.tools.ietf.org" no longer exists.

**Comment by abierman** [comment]
why is this complexity:unknown?
The usage is rather simple for the user.
It is the backward compatibility that is the source of the complexity

From 7.8.2:
   All key leafs MUST be given values
   when a list entry is created.  Thus, any default values in the key
   leafs or their types are ignored.

It may not be wise to change the key-stmt semantics to overturn a MUST.
Instead a new statement could be used (optkey or key2?)

**Comment by mbj4668** [comment]
I agree some new statement is needed.  Either mark the optional leafs in the definition of the keys:
    key "foo bar? baz?";
or in the leaf definition:
   key "foo bar baz";
   ...
   leaf bar {
     optional-key true;
     ...
   }

**Comment by immibis** [comment]
Rather than `optional-key true;` I would expect it to be `default "whatever";`. The proposal is for keys to have default values, not for keys to be fully optional.

**Comment by kwatsen** [comment]
Jan: how to specify the value if not passed?
Andy: could/should use the "default" value
Kent: what is the use-case?
Everyone: not importance high, so moving to medium
